flink-clients/src/main/java/org/apache/flink/client/program/ClusterClient.java

```diff
@@ -684,6 +688,29 @@ public void stop(final JobID jobId) throws Exception {
 	}
 
 	/**
+	 * Lists the currently running jobs on the cluster.
+	 *
+	 * @return future collection of running jobs
+	 * @throws Exception if  no connection to the cluster could be established
+	 */
+	public CompletableFuture<Collection<JobDetails>> listJobs() throws Exception {
```

Comment:
I'm wondering whether the `ClusterClient` should issue a synchronous operation instead of a asynchronous operation (simply waiting on the futures completion). So far all the `ClusterClient` operations are blocking and wait for the completion. Not entirely sure which is more intuitive to use for the user. Of course the asynchronous variant gives more freedom.